analysis_options.yaml

```diff
@@ -2,6 +2,8 @@
 include: package:dart_flutter_team_lints/analysis_options.yaml
 
 analyzer:
+  enable-experiment:
+    - inline-class
```

Comment:
I'd likely just update the SDK constraint to `^3.3.0-214.0.dev` or similar instead, which should not require this and is the first public `-dev` release with https://github.com/dart-lang/sdk/commit/b1a7ca77e0272071f41d9eeaadadc52906a1dbc9.

Reply:
I updated the CI to only test `beta` and `dev`, which I believe should both have that version. I'm not sure if we want to update the SDK constraint or not, but I'm leaning towards yes. The benefit of not doing so is users on 3.2 can use this new version, but maybe that's not something we want anyways.

Reply:
Since we added the package into an allow list to be able to use it early, we can use 3.3.0-76.0.dev (see https://github.com/dart-lang/sdk/commit/f900eb95b784968d04a34bc5f8eeb83afc713aa9), still not a 3.2 unfortuantely.